### PR TITLE
[api] Adding contact feature.

### PR DESCRIPTION
**Pull request by MohammedAlewi**

### CODE REVIEW

File: Backend/models/ContactModel.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const contactSchema = new mongoose.Schema({
+    _id: mongoose.Schema.Types.ObjectId,
+
+    name:{
+        type: String,
+        required: true,
+    },
+    email:{
+        type: String,
```
Comment (Khalid-Sultan) [inline Backend/models/ContactModel.js:12]: Email and the other fields are also required by default

File: Backend/controllers/ContactController.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+// update a contact
+exports.update_contact = async (req, res) => {
+    let contact = await Contact.findById(req.params.id);
+    if (!contact) {
+        return res.status(404).send("Contact doesn't exist.s");
+    }
+    try {
+        contact = await Contact.findByIdAndUpdate(req.params.id,req.body,{new: true})
```
Comment (Khalid-Sultan) [inline Backend/controllers/ContactController.js:81]: Validating the body before hand might be a better practice for safety.

File: Backend/controllers/ContactController.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+};
+
+
+// Deleting a contact
+exports.delete_contact = async (req, res) => {
+    try {
+        let contact = await Contact.findByIdAndRemove(req.params.id);
+        if (!contact) {
+            res.status(404).send("contact not found");
```
Comment (Khalid-Sultan) [inline Backend/controllers/ContactController.js:94]: Adding 'return' would prevent the code from executing the next line

File: Backend/controllers/ContactController.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    );
+
+    let result = {
+        data_count: await Contact.countDocuments(filter),
+        page_size: size,
+        current_page: page,
+        data: contacts,
+    };
+
+    try {
```
Comment (AntenehAdmasu) [inline Backend/controllers/ContactController.js:38]: Lets remove try/catch blocks from now on. Let's handle them centrally.

**Review by Khalid-Sultan — APPROVED** [review]
Looks good to me :+1:

File: Backend/controllers/ContactController.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        Object.assign(filter,{subject : { $regex: re }})
+    }
+
+
+    let page = parseInt(req.query.page) || 1;
+    let size = parseInt(req.query.size) || 15;
+
+    const contacts = await Contact.find(
+        filter,{},
+        { skip: (page - 1) * size, limit: size * 1 }
```
Comment (AntenehAdmasu) [inline Backend/controllers/ContactController.js:28]: we can centralize the paginations too :)

File: Backend/controllers/ContactController.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    return res.status(201).send(contact);
+};
+
+
+// update a contact
+exports.update_contact = async (req, res) => {
+    let contact = await Contact.findById(req.params.id);
+    if (!contact) {
+        return res.status(404).send("Contact doesn't exist.s");
```
Comment (AntenehAdmasu) [inline Backend/controllers/ContactController.js:69]: seen this typo? 😁

**Pull request closed without merging**